src/bundle/Controller/BookmarkController.php

```diff
@@ -68,17 +68,19 @@ public function __construct(
      * @param \Symfony\Component\HttpFoundation\Request $request
      *
      * @return \Symfony\Component\HttpFoundation\Response
-     *
-     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
      */
     public function listAction(Request $request): Response
     {
         $page = $request->query->get('page', 1);
-        $bookmarkDataset = $this->datasetFactory->bookmarks();
-        $bookmarkDataset->load();
 
         $pagerfanta = new Pagerfanta(
-            new ArrayAdapter($bookmarkDataset->getBookmarks())
+            new CallbackAdapter(function () {
```

Comment:
IMHO this might be extracted to separate class:  https://gist.github.com/adamwojs/d90ac7ce37e91ee5c0fe45797a58793a